src/box/lua/load_cfg.lua

```diff
@@ -219,40 +219,17 @@ local template_cfg = {
     txn_timeout           = 'number',
 }
 
-local function normalize_uri(port)
-    if port == nil or type(port) == 'table' then
-        return port
-    end
-    return tostring(port);
-end
-
 local function normalize_uri_list_for_replication(port_list)
-    local result = {}
     if type(port_list) == 'table' then
-        for _, port in ipairs(port_list) do
-            table.insert(result, normalize_uri(port))
-        end
-    else
-        table.insert(result, normalize_uri(port_list))
+        return port_list
     end
-    return result
-end
-
-local function normalize_uri_list_for_listen(port_list)
     local result = {}
-    if type(port_list) == 'table' then
-        for _, port in ipairs(port_list) do
-            table.insert(result, normalize_uri(port))
-        end
-    else
-        return normalize_uri(port_list)
-    end
+    table.insert(result, port_list)
     return result
 end
 
 -- options that require special handling
 local modify_cfg = {
-    listen             = normalize_uri_list_for_listen,
     replication        = normalize_uri_list_for_replication,
```

Comment:
Can't you drop `normalize_uri_list_for_replication`? AFAIU `luaT_uri_set_create` handles one URI just as well, no?

Reply:
I wan't to do it. But box.cfg.replication should be a table even in case when replication is a single string! So problem is not in `luaT_uri_set_create `.

Reply:
Why does it have to be a table?

If I set `box.cfg.replication` to `{'uri', params = {param1 = 'abc'}}`, technically it will be a table, but it won't be an array of URIs so this normalization doesn't actually work.

Reply:
It is necessary to prevent breaking backwards compatibility:
box.cfg { replication = "/tmp/1" } should be a table after normalization. All of our tests rely on it. Maybe some users also.

Reply:
Okay. Let's leave it as is then.